src/site/stages/build/plugins/create-resources-and-support-section.js

```diff
@@ -291,44 +291,52 @@ function createArticleListingsPages(files) {
   );
 }
 
-function createArticleResultSnippet(text) {
-  const sanitizedText = liquid.filters.strip_html(text);
+function deriveIntroText(htmlText, options = {}) {
+  const sanitizedText = liquid.filters.strip_html(htmlText);
   const strippedNewlines = liquid.filters.strip_newlines(sanitizedText);
   const decodedText = he.decode(strippedNewlines);
 
+  // Truncate the text.
   const truncated = liquid.filters.truncate(decodedText, 190);
 
-  if (decodedText !== truncated) {
-    return `${truncated}...`;
+  // Return full html escaped text if we are not truncating text or if text is brief.
+  if (!options?.truncated || decodedText === truncated) {
+    return decodedText;
   }
 
-  return truncated;
+  // If the text is long, show an ellipsis at the end of it.
+  return `${truncated}...`;
 }
 
 function createSearchResults(files) {
   const allArticles = getArticlesBelongingToResourcesAndSupportSection(files);
   const articleSearchData = allArticles.map(article => {
-    let limitedDescription = null;
+    let truncatedIntroText = '';
+    let introText = '';
 
     if (article.entityBundle === 'q_a') {
       const answer = article.fieldAnswer.entity.fieldWysiwyg.processed;
-      limitedDescription = createArticleResultSnippet(answer);
+      truncatedIntroText = deriveIntroText(answer, { truncated: true });
+      introText = deriveIntroText(answer, { truncated: false });
     } else {
-      limitedDescription = createArticleResultSnippet(
+      truncatedIntroText = deriveIntroText(
         article.fieldIntroTextLimitedHtml.processed,
+        { truncated: true },
       );
+      introText = deriveIntroText(article.fieldIntroTextLimitedHtml.processed, {
+        truncated: false,
+      });
     }
 
     return {
       entityBundle: article.entityBundle,
-      entityUrl: {
-        path: article.entityUrl.path,
-      },
-      title: article.title,
-      description: limitedDescription,
-      fieldPrimaryCategory: article.fieldPrimaryCategory,
+      entityUrl: { path: article.entityUrl.path },
       fieldOtherCategories: article.fieldOtherCategories,
+      fieldPrimaryCategory: article.fieldPrimaryCategory,
       fieldTags: article.fieldTags,
+      introText,
+      title: article.title,
+      truncatedIntroText,
```

Comment:
By how much does including the full intro text increase the file size?

Reply:
File size went from 31kb to 44kb for `articles.json`. This is something I hadn't really thought about, but if we were to add on that `searchableContent` field for each article that is going to make the file size explode here 🤔

Reply:
That said 44kb doesn't seem like that much more than 31kb for this PR

Reply:
Good news! Managed to get it down to 34kb by removing the unnecessary `truncatedIntroText` field, which we can just do with lodash's `truncate` method when rendering an article 👍

Reply:
Very cool! With React, do we need to worry about that issue w/html entities getting truncated in the middle?

Reply:
`introText` is not html anymore since we're using `he` to sanitize it 🙂